python/ray/rllib/models/preprocessors.py

```diff
@@ -140,6 +140,9 @@ def _init_shape(self, obs_space, options):
 
     @override(Preprocessor)
     def transform(self, observation):
+        if not self._obs_space.contains(observation):
```

Comment:
I wonder what the performance overhead of this is -- there's a unit test in test_policy_evaluator.py called "testBaselinePerformance", could you try reporting before/after samples per second?

Reply:
It seems that the effort is not that light. samples per second drops from 4296.390228470533 to 3326.3202875529773.